### PR TITLE
Introduce Bucket controller for Asset Store

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Introduce Bucket controller for Asset Store

**Related issue(s)**
See #2269

### CODE REVIEW

File: components/assetstore-controller-manager/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
```
Comment (pkosiec) [inline components/assetstore-controller-manager/Makefile]: The Makefile has to be adjusted. And we probably need `before-commit.sh`, don't we?

**Comment by pkosiec** [comment]
Oh, I forgot to update CODEOWNERS. I will do that after review 🙂

File: components/assetstore-controller-manager/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+WORKDIR /go/src/github.com/kyma-project/kyma/components/bucket-controller
+COPY pkg/    pkg/
+COPY cmd/    cmd/
+COPY vendor/ vendor/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager github.com/kyma-project/kyma/components/bucket-controller/cmd/manager
+
+# Copy the controller-manager into a thin image
+FROM ubuntu:latest
```
Comment (crabtree) [inline components/assetstore-controller-manager/Dockerfile]: Please consider using a different base image (like alpine or scratch) and definitely do not use the latest tag.

File: components/assetstore-controller-manager/pkg/controller/bucket/bucket_controller.go [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+	}
+
+	if !exists && instance.Status.Reason == ReasonNotFound {
+		log.Info(fmt.Sprintf("Bucket %s still doesn't exist. Scheduling requeue...", bucketName))
+		return reconcile.Result{RequeueAfter: r.requeueInterval}, nil
+	}
+
+	if !exists {
+		// Bucket was created before, but has been deleted manually
+		log.Info(fmt.Sprintf("Bucket %s/%s has been deleted. Setting failed status...", instance.Namespace, instance.Name))
```
Comment (michal-hudy) [inline components/assetstore-controller-manager/pkg/controller/bucket/bucket_controller.go]: Warning?

File: components/assetstore-controller-manager/pkg/buckethandler/buckethandler.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	err = h.client.MakeBucket(bucketName, region)
+	if err != nil {
+		return false, errors.Wrapf(err, "while creating bucket %s in region %s", bucketName, region)
+	}
+	h.logInfof("Bucket %s created in region %s", bucketName, region)
+
+	return true, nil
+}
+
+func (h *bucketHandler) CheckIfExists(bucketName string) (bool, error) {
```
Comment (michal-hudy) [inline components/assetstore-controller-manager/pkg/buckethandler/buckethandler.go]: Maybe just `Exists`?

File: components/assetstore-controller-manager/Makefile (hunk 1 above)
Comment (pkosiec) [reply components/assetstore-controller-manager/Makefile]: Okay then, so I won't touch it here 😄

**Pull request merged**